lib/graphql/schema.rb

```diff
@@ -444,13 +445,22 @@ def self.from_introspection(introspection_result)
       GraphQL::Schema::Loader.load(introspection_result)
     end
 
-    # Create schema from an IDL schema.
-    # @param definition_string [String] A schema definition string
+    # Create schema from an IDL schema or file containing an IDL definition.
+    # @param definition_or_path [String] A schema definition string, or a path to a file containing the definition
     # @param default_resolve [<#call(type, field, obj, args, ctx)>] A callable for handling field resolution
     # @param parser [Object] An object for handling definition string parsing (must respond to `parse`)
     # @return [GraphQL::Schema] the schema described by `document`
-    def self.from_definition(string, default_resolve: BuildFromDefinition::DefaultResolve, parser: BuildFromDefinition::DefaultParser)
-      GraphQL::Schema::BuildFromDefinition.from_definition(string, default_resolve: default_resolve, parser: parser)
+    def self.from_definition(definition_or_path, helpers: nil, default_resolve: BuildFromDefinition::DefaultResolve, parser: BuildFromDefinition::DefaultParser)
+      # If the file ends in `.graphql`, treat it like a filepath
+      definition = if definition_or_path.end_with?(".graphql")
+        File.read(definition_or_path)
+      elsif definition_or_path.end_with?(".erb")
+        erb_template = File.read(definition_or_path)
+        GraphQL::Schema::Template.run(erb_template, helpers: helpers)
+      else
+        definition_or_path
+      end
+      GraphQL::Schema::BuildFromDefinition.from_definition(definition, default_resolve: default_resolve, parser: parser)
```

Comment:
Maybe this could support an _array_ of definitions/paths. That way you could split your schema up into several files and pass all of them in together:
```ruby
def self.from_definition(definition_or_path, helpers: nil, default_resolve: BuildFromDefinition::DefaultResolve, parser: BuildFromDefinition::DefaultParser)
  definition_or_path = [definition_or_path] unless definition_or_path.is_a?(Array)

  schemas = definition_or_path.map do |value|
    # If the value ends in `.graphql` or `.erb`, treat it like a filepath
    if value.end_with?('.graphql')
      File.read(value)
    elsif value.end_with?('.erb')
      erb_template = File.read(value)
      GraphQL::Schema::Template.run(erb_template, helpers: helpers)
    else
      value
    end
  end

  GraphQL::Schema::BuildFromDefinition.from_definition(schemas.join("\n"), default_resolve: default_resolve, parser: parser)
end
```

Example:
```ruby
GraphQL::Schema.from_definition(Dir["schema/**/*.graphql", "schema/**/*.graphql.erb"])
```